addons/hr_attendance/models/res_users.py

```diff
@@ -24,3 +25,15 @@ def SELF_READABLE_FIELDS(self):
             'last_check_out',
             'total_overtime'
         ]
+
+    def _clean_attendance_officers(self):
+        officer_group = 'hr_attendance.group_hr_attendance_officer'
+        res = self.env['hr.employee']._read_group(
+            [('attendance_manager_id', 'in', self.ids)],
+            ['attendance_manager_id'])
+
+        officers_to_remove_ids = set(self.ids) - {att_officer.id for [att_officer] in res}
```

Comment:
I don't think here you need to do set  :thinking: 
```suggestion
        officers_to_remove_ids = self.ids - {attendance_manager.id for [attendance_manager] in attendance_manager_ids}
```

Reply:
I don't need to, but i think it's more efficient when doing difference here